suite-native/module-accounts-import/src/components/XpubHintModal.tsx

```diff
@@ -0,0 +1,114 @@
+import { Icon } from '@trezor/icons';
+import { NetworkType } from '@trezor/connect/lib/constants/network';
+import { Video, VideoName } from '@suite-native/video-assets';
+
+type XpubScanHintSheet = {
+    networkType: NetworkType;
+};
+
+const sheetTriggerStyle = prepareNativeStyle(_ => ({
+    flexDirection: 'row',
+    width: '100%',
+    justifyContent: 'center',
+    alignItems: 'center',
+}));
+
+const EmphasizedText = ({ children }: { children: ReactNode }) => (
+    <Text color="gray900" variant="hint">
+        {children}
+    </Text>
+);
+
+type NetworkAssets = {
+    title: string;
+    text: ReactNode;
+    video: VideoName;
+};
+
+const networkToAssetsMap = {
+    bitcoin: {
+        title: 'Where is my public key (XPUB)?',
+        text: (
+            <>
+                To view the Bitcoin XPUB of your account, open the Trezor Suite desktop app, plug in
+                your Trezor device, select <EmphasizedText>Accounts</EmphasizedText>, then choose{' '}
+                <EmphasizedText>Show public key</EmphasizedText>.
+            </>
+        ),
+        video: 'xpubImportBTC',
+    },
+    ethereum: {
+        title: 'Where is my receive address?',
+        text: (
+            <>
+                To view the Ethereum and Ethereum Classic receive address of your account, open the
+                Trezor Suite desktop app, plug in your Trezor device, select{' '}
+                <EmphasizedText>Accounts</EmphasizedText>, choose Receive, and click on{' '}
+                <EmphasizedText>Show full address</EmphasizedText>.
+            </>
+        ),
+        video: 'xpubImportETH',
+    },
+    ripple: {
+        title: 'Where is my receive address?',
+        text: (
+            <>
+                To view the Ripple receive address of your account, open the Trezor Suite desktop
+                app, plug in your Trezor device, select <EmphasizedText>Accounts</EmphasizedText>,
+                choose
+                <EmphasizedText>Receive</EmphasizedText>, and click on{' '}
+                <EmphasizedText>Show full address</EmphasizedText>
+            </>
+        ),
+        video: 'xpubImportETH',
+    },
+    eos: undefined,
+    nem: undefined,
+    stellar: undefined,
+    cardano: undefined,
+    tezos: undefined,
+    binance: undefined,
+} as const satisfies Record<NetworkType, NetworkAssets | undefined>;
+
+const defaultAssets = networkToAssetsMap.bitcoin;
+
+export const XpubHintModal = ({ networkType }: XpubScanHintSheet) => {
+    const [isVisible, setIsVisible] = useState(false);
+    const { applyStyle } = useNativeStyles();
+
+    const networkAssets = networkToAssetsMap[networkType];
+    const { title, text, video } = networkAssets ?? defaultAssets;
+
+    const handleOpen = () => setIsVisible(true);
+    const handleClose = () => setIsVisible(false);
+
+    return (
+        <View>
+            <TouchableOpacity onPress={handleOpen} style={applyStyle(sheetTriggerStyle)}>
```

Comment:
Please add todo comment here that this should be replaced with `TextButton` once we will have it.

Reply:
fixed 781b757df